test/buy-tshirt.spec.ts

```diff
@@ -17,25 +17,38 @@ describe('Buy a t-shirt', () => {
   const orderSummaryPage: OrderSummaryPage = new OrderSummaryPage();
   const paymentStepPage: PaymentStepPage = new PaymentStepPage();
   const productAddedModalPage: ProductAddedModalPage = new ProductAddedModalPage();
-  const productListPage: ProductListPage = new ProductListPage();
   const shippinStepPage: ShippinStepPage = new ShippinStepPage();
   const signInStepPage: SignInStepPage = new SignInStepPage();
   const summaryStepPage: SummaryStepPage = new SummaryStepPage();
+  const productAddedPage: ProductAddedPage = new ProductAddedPage();
 
-  it('then should be bought a t-shirt', async () => {
-    await browser.get('http://automationpractice.com/');
-    await menuContentPage.goToTShirtMenu();
-    await addressStepPage.addToCar();
-    await bankPaymentPage.proceedToCheckout();
-    await orderSummaryPage.proceedToCheckout();
-    await paymentStepPage.fillEmailPassword();
-    await paymentStepPage.registerEmail();
-    await productAddedModalPage.proceedToCheckout();
-    await shippinStepPage.selectCheckBoxAgreeTerms();
-    await productListPage.proceedToCheckout();
-    await signInStepPage.payByBankWire();
-    await summaryStepPage.confirmOrder();
-    await expect(summaryStepPage.mensajeText())
-      .toBe('Your order on My Store is complete.');
+  describe('then should be bought a t-shirt', () => {
+
+    beforeEach(async () => {
+      await browser.get('http://automationpractice.com/');
+    });
+
+    describe('this execute all the lines into the beforeEach', () => {
+
+      beforeEach(async () => {
+        await menuContentPage.goToTShirtMenu();
+        await addressStepPage.selectProduct();
+        await productAddedPage.addToCar();
+        await bankPaymentPage.proceedToCheckout();
+        await orderSummaryPage.proceedToCheckout();
+        await paymentStepPage.fillEmailPassword();
+        await paymentStepPage.registerEmail();
+        await productAddedModalPage.proceedToCheckout();
+        await shippinStepPage.selectCheckBoxAgreeTerms();
+        await shippinStepPage.proceedToCheckout();
+        await signInStepPage.payByBankWire();
+        await summaryStepPage.confirmOrder();
+      });
+
+      it('Verify that your order is complete', async () => {
+        await expect(summaryStepPage.mensajeText())
```

Comment:
El expect no necesita await, lo que está dentro del expect sí, quedaría algo cómo:
expect(await summaryStepPage.mesageText()).toBe('Your order on My Store is complete.')

Reply:
corregido en buy-tshirt.spec